app/src/main/java/com/uncookthebook/app/ReportArticleFragment.java

```diff
@@ -2,72 +2,124 @@
 
+import lombok.SneakyThrows;
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
 
 public class ReportArticleFragment extends Fragment {
 
     private static final String TAG = "ReportArticleFragment";
 
-    private String url;
+    private URL url;
     private ZoomOnClick animationHandler = new ZoomOnClick();
+    private View view;
 
     @Override
     public View onCreateView(
             @NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
         // Inflate the layout for this fragment with the ProductGrid theme
-        View view = inflater.inflate(R.layout.fragment_report_article, container, false);
-        voteButtonSetup(view, R.id.button_legit);
-        voteButtonSetup(view, R.id.button_fake);
-        homeButtonSetup(view);
+        view = inflater.inflate(R.layout.fragment_report_article, container, false);
+        voteButtonSetup(R.id.button_legit);
+        voteButtonSetup(R.id.button_fake);
+        homeButtonSetup();
         return view;
     }
 
     @Override
     public void onStart() {
         super.onStart();
         getURL();
+        setSiteLogo(url.getProtocol() + "://" + url.getHost() + "/favicon.ico");
+        setSiteTitle();
     }
 
+    @SneakyThrows
     private void getURL() {
         final SharedPreferences sharedPref = Objects.requireNonNull(getContext()).getSharedPreferences(
                 getString(R.string.preference_file_key),
                 Context.MODE_PRIVATE
         );
-        url = sharedPref.getString(getString(R.string.url_key), null);
-        if(url != null) {
+        String urlString = sharedPref.getString(getString(R.string.url_key), null);
+        if(urlString != null) {
             SharedPreferences.Editor editor = sharedPref.edit();
             editor.remove(getString(R.string.url_key));
             editor.apply();
-            Log.d("URL", url);
+            Log.d("URL", urlString);
+            this.url = new URL(urlString);
         }
     }
 
-    private void voteButtonSetup(View view, int id){
+    private void voteButtonSetup(int id){
         MaterialButton button = view.findViewById(id);
         animationHandler.addView(button);
         button.setOnClickListener(v -> {
             animationHandler.handleSize(v, getContext());
         });
     }
 
-    private void homeButtonSetup(View view){
+    private void homeButtonSetup(){
         MaterialButton button = view.findViewById(R.id.button_home);
         button.setOnClickListener(v -> {
             ((NavigationHost) Objects.requireNonNull(getActivity())).navigateTo(
                     new PasteArticleFragment(), false
             );
         });
     }
+
+    private void setSiteLogo(String src) {
+        final String TAG = "SITE_LOGO_RETRIEVAL";
+        final OkHttpClient okHttpClient = new OkHttpClient();
+        final Request request = new Request.Builder().url(src).build();
```

Comment:
Maybe we should use APIService to manage the request instead of creating a new client